### ISSUE TITLE
Ch2 end-to-end example does not improve in accuracy

### ISSUE THREAD

**Issue by ChosunOne** [PacktPublishing/What-s-New-in-TensorFlow-2.0#1]
On running the example for Chapter 2's end to end example, the accuracy remains at approximately 11% and does not improve.

**Comment by ChosunOne** [comment]
For accuracy to improve the values in x_train and x_test need to be divided by 255

**Comment by alye** [comment]
While scaling images is helpful (and recommended in the real world), the goal of the example is to show how to build a model using the Sequential API, not necessarily achieve state of the art performance. The scaling step has been omitted from the book for the sake of conciseness.

What hardware / TF version are you using? It seems to be working fine without scaling at my end. (See attached screenshot)

Could you try running it in its original form again? 

![model_train](https://user-images.githubusercontent.com/11714174/65134345-00605080-d9b9-11e9-985a-160a91167625.png)

**Comment by ChosunOne** [comment]
I'm using an Nvidia 1080 Ti with tensorflow release candidate 2.0.0rc0.  Running the original code does not improve the model at all, but scaling the values fixes it entirely.

**Comment by ChosunOne** [comment]
The issue persists after upgrading to tensorflow 2.0.0rc1

**Comment by ChosunOne** [comment]
```
  32/60000 [..............................] - ETA: 3s - loss: 2.3309 - accuracy: 0.0312
 1248/60000 [..............................] - ETA: 2s - loss: 2.3027 - accuracy: 0.1018
 2400/60000 [>.............................] - ETA: 2s - loss: 2.3032 - accuracy: 0.0975
 3552/60000 [>.............................] - ETA: 2s - loss: 2.3021 - accuracy: 0.1028
 4704/60000 [=>............................] - ETA: 2s - loss: 2.3013 - accuracy: 0.1084
 5888/60000 [=>............................] - ETA: 2s - loss: 2.3016 - accuracy: 0.1084
 7104/60000 [==>...........................] - ETA: 2s - loss: 2.3017 - accuracy: 0.1087
 8288/60000 [===>..........................] - ETA: 2s - loss: 2.3014 - accuracy: 0.1100
 9568/60000 [===>..........................] - ETA: 2s - loss: 2.3012 - accuracy: 0.1110
10816/60000 [====>.........................] - ETA: 2s - loss: 2.3013 - accuracy: 0.1107
12000/60000 [=====>........................] - ETA: 2s - loss: 2.3015 - accuracy: 0.1107
13184/60000 [=====>........................] - ETA: 1s - loss: 2.3017 - accuracy: 0.1092
14400/60000 [======>.......................] - ETA: 1s - loss: 2.3018 - accuracy: 0.1082
15584/60000 [======>.......................] - ETA: 1s - loss: 2.3018 - accuracy: 0.1084
16800/60000 [=======>......................] - ETA: 1s - loss: 2.3018 - accuracy: 0.1085
18048/60000 [========>.....................] - ETA: 1s - loss: 2.3017 - accuracy: 0.1097
19328/60000 [========>.....................] - ETA: 1s - loss: 2.3016 - accuracy: 0.1102
20448/60000 [=========>....................] - ETA: 1s - loss: 2.3017 - accuracy: 0.1096
21536/60000 [=========>....................] - ETA: 1s - loss: 2.3016 - accuracy: 0.1103
22656/60000 [==========>...................] - ETA: 1s - loss: 2.3014 - accuracy: 0.1114
23872/60000 [==========>...................] - ETA: 1s - loss: 2.3014 - accuracy: 0.1115
25088/60000 [===========>..................] - ETA: 1s - loss: 2.3014 - accuracy: 0.1115
26272/60000 [============>.................] - ETA: 1s - loss: 2.3015 - accuracy: 0.1110
27520/60000 [============>.................] - ETA: 1s - loss: 2.3014 - accuracy: 0.1116
28640/60000 [=============>................] - ETA: 1s - loss: 2.3013 - accuracy: 0.1121
29856/60000 [=============>................] - ETA: 1s - loss: 2.3014 - accuracy: 0.1122
31040/60000 [==============>...............] - ETA: 1s - loss: 2.3013 - accuracy: 0.1126
32224/60000 [===============>..............] - ETA: 1s - loss: 2.3012 - accuracy: 0.1126
33408/60000 [===============>..............] - ETA: 1s - loss: 2.3011 - accuracy: 0.1130
34560/60000 [================>.............] - ETA: 1s - loss: 2.3012 - accuracy: 0.1131
35808/60000 [================>.............] - ETA: 1s - loss: 2.3012 - accuracy: 0.1128
36992/60000 [=================>............] - ETA: 0s - loss: 2.3012 - accuracy: 0.1131
38016/60000 [==================>...........] - ETA: 0s - loss: 2.3012 - accuracy: 0.1127
39200/60000 [==================>...........] - ETA: 0s - loss: 2.3013 - accuracy: 0.1123
40288/60000 [===================>..........] - ETA: 0s - loss: 2.3013 - accuracy: 0.1122
41472/60000 [===================>..........] - ETA: 0s - loss: 2.3013 - accuracy: 0.1126
42624/60000 [====================>.........] - ETA: 0s - loss: 2.3013 - accuracy: 0.1123
43744/60000 [====================>.........] - ETA: 0s - loss: 2.3013 - accuracy: 0.1124
44800/60000 [=====================>........] - ETA: 0s - loss: 2.3013 - accuracy: 0.1123
45984/60000 [=====================>........] - ETA: 0s - loss: 2.3013 - accuracy: 0.1126
47168/60000 [======================>.......] - ETA: 0s - loss: 2.3013 - accuracy: 0.1126
48320/60000 [=======================>......] - ETA: 0s - loss: 2.3013 - accuracy: 0.1126
49376/60000 [=======================>......] - ETA: 0s - loss: 2.3013 - accuracy: 0.1125
50624/60000 [========================>.....] - ETA: 0s - loss: 2.3013 - accuracy: 0.1123
51904/60000 [========================>.....] - ETA: 0s - loss: 2.3014 - accuracy: 0.1122
53120/60000 [=========================>....] - ETA: 0s - loss: 2.3014 - accuracy: 0.1121
54304/60000 [==========================>...] - ETA: 0s - loss: 2.3013 - accuracy: 0.1121
55424/60000 [==========================>...] - ETA: 0s - loss: 2.3013 - accuracy: 0.1121
56608/60000 [===========================>..] - ETA: 0s - loss: 2.3013 - accuracy: 0.1122
57888/60000 [===========================>..] - ETA: 0s - loss: 2.3013 - accuracy: 0.1124
59008/60000 [============================>.] - ETA: 0s - loss: 2.3013 - accuracy: 0.1124
60000/60000 [==============================] - 3s 43us/sample - loss: 2.3013 - accuracy: 0.1124
```

That is what I see for the final epoch.  Accuracy remains at 11% the entire time.

**Comment by ChosunOne** [comment]
Running the exact code on GitHub it does even worse, with only 6% training accuracy:
```
   32/60000 [..............................] - ETA: 6:44 - loss: 134.3741 - accuracy: 0.0625Epoch 2/10
   32/60000 [..............................] - ETA: 4s - loss: 4.6464 - accuracy: 0.0938Epoch 3/10
   32/60000 [..............................] - ETA: 3s - loss: 2.3026 - accuracy: 0.1250Epoch 4/10
   32/60000 [..............................] - ETA: 3s - loss: 11.4289 - accuracy: 0.0625Epoch 5/10
   32/60000 [..............................] - ETA: 3s - loss: 5.8086 - accuracy: 0.1562Epoch 6/10
   32/60000 [..............................] - ETA: 3s - loss: 2.3017 - accuracy: 0.0625Epoch 7/10
   32/60000 [..............................] - ETA: 3s - loss: 2.3044 - accuracy: 0.0625Epoch 8/10
   32/60000 [..............................] - ETA: 3s - loss: 2.3030 - accuracy: 0.0625Epoch 9/10
   32/60000 [..............................] - ETA: 4s - loss: 2.3027 - accuracy: 0.0625Epoch 10/10
```

**Comment by alye** [comment]
I have been able to reproduce the issue at my end on the CPU version as well. However the code in its original form continues to work well with  the ```2.0.0-beta``` release. 

It might be worth filing a bug report on the Tensorflow repository as this is inconsistent behavior between two releases.

**Comment by ChosunOne** [comment]
It seems that the behavior in beta 0 was not the intended behavior.  Changing the learning rate to a smaller value (0.0005) is said to also resolve the issue.

**Comment by sarvesh-packt** [comment]
Hi, 

Please reopen if the issue still persists.

Thanks,
